src/dev/flang/ast/If.java

```diff
@@ -407,20 +375,42 @@ public void propagateExpectedType(Resolution res, AbstractFeature outer)
    */
   public Expr propagateExpectedType(Resolution res, AbstractFeature outer, AbstractType t)
   {
+    if (PRECONDITIONS) require
+      (elseBlock != null || elseIf != null);
+
     return addFieldForResult(res, outer, t);
   }
 
 
   /**
-   * Get (or create empty) else block for this if.
+   * If there is no else / elseif, create a default else
+   * branch returning unit.
+   */
+  private void CreateDefaultElseIfMissing()
+  {
+    if (elseBlock == null && elseIf == null)
+      {
+        var unit = new Call(pos(), "unit");
+        elseBlock = new Block(pos(), new List<Stmnt>(unit));
+      }
+  }
+
+
+  /**
+   * Get else block for this if.
    *
    * @return an else block.
    */
   public Expr elseBlock()
   {
-    return
-      elseBlock != null ? elseBlock :
-      elseIf    != null ? elseIf    : new Block(pos(), new List<>());
+    if (PRECONDITIONS) require
+      (elseBlock != null || elseIf != null);
+
+    if (elseBlock != null)
+      {
+        return elseBlock;
+      }
+    return elseIf;
```

Comment:
No a fan of `return` in nested blocks, use 
```
return 
  elseBlock != null ? elseBlock 
                    : elseIf;
```
instead (even if `? :` for many developers needs some time to get used to...).